modin/pandas/indexing.py

```diff
@@ -700,7 +701,10 @@ def __setitem__(self, key, item):
             isinstance(row_loc, list)
             and len(row_loc) == 1
             and row_loc[0] not in self.qc.index
-        ) or (not isinstance(row_loc, list) and row_loc not in self.qc.index):
+        ) or (
+            not isinstance(row_loc, (collections.abc.Sized, slice))
```

Comment:
Would this be the same? I prefer to avoid adding `abc.Sized` here 
```suggestion
            not is_list_like(row_loc)
```

Reply:
`is_list_like` on a slice returns `False`

Reply:
```suggestion
            not is_list_like(row_loc) or isinstance(row_loc, slice)
```